internal/juju/client.go

```diff
@@ -134,6 +134,9 @@ type sharedClient struct {
 
 	checkJAASOnce sync.Once
 	isJAAS        bool
+
+	externalControllerConfigs map[string]ControllerConfiguration
+	lockExtCtlrMap            sync.RWMutex
```

Comment:
can we call it `externalControllerConfigsMu` so it's obvious from the name, which field this is protecting.. also agree with Kian about RWMutex.. and name, right.. should be `offeringControllerConfigs` :)